languages/csharp/exercises/concept/basics/.docs/introduction.md

```diff
@@ -0,0 +1,57 @@
+C# is a statically-typed language, which means that everything has a type at compile-time. Assigning a value to a name is referred to as defining a variable. A variable can be defined either by explicitly specifying its type, or by letting the C# compiler infer its type based on the assigned value (known as _type inference_). Therefore, the following two variable definitions are equivalent:
+
+```csharp
+int fingers = 10;
+var fingers = 10;
+```
+
+Updating a variable's value is done through the `=` operator:
+
+```csharp
+var count = 5;
+count = 7;
+```
+
+C# is an object-oriented language and requires all functions to be defined in a _class_. The `class` keyword is used to define a class. It is common to specify an _access modifier_, which influences what other classes can use the class. The most common access modifier is `public`, which means that other classes can use this class.
+
+```csharp
+public class Calculator
+{
+}
+```
+
+A function within a class is referred to as a _method_. Each method can have zero or more parameters. All parameters must be explicitly typed, there is no type inference for parameters. Similarly, the return type must also be made explicit. Like classes, an access modifier can be specified for methods. Values are returned from functions using the `return` keyword.
+
+```csharp
+public int Add(int x, int y)
+{
+    return x + y;
+}
+```
+
+Invoking a method is done by specifying its class- and method name and passing arguments for each of the method's parameters.
+
+```csharp
+public int AddToSelf(int x)
+{
+    return Calculator.Add(x, x);
+}
+```
+
+If the method to be called is defined in the same class as the method that calls it, the class name can be omitted.
+
+```csharp
+public int AddToSelf(int x)
+{
+    return Add(x, x);
+}
+```
+
+C# supports two types of comments. Single line comments are preceded by `//` and multiline comments are inserted between `/*` and `*/`.
+
+```csharp
```

Comment:
I don't think we need this example, as they'll be visible in the stub?

Reply:
The multi-line comment won't be, but I agree that the example is overkill. I've removed the example and added a link in the `after.md` document.